src/main/java/de/hhu/propra/sharingplatform/service/ApiService.java

```diff
@@ -0,0 +1,105 @@
+package de.hhu.propra.sharingplatform.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.hhu.propra.sharingplatform.dao.PaymentRepo;
+import de.hhu.propra.sharingplatform.dto.ProPay;
+import de.hhu.propra.sharingplatform.dto.ProPayReservation;
+import de.hhu.propra.sharingplatform.model.Payment;
+import de.hhu.propra.sharingplatform.model.User;
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+@Data
+@Service
+public class ApiService {
+
+    private PaymentRepo paymentRepo;
+    String host = "localhost";
+
+    @Autowired
+    public ApiService(PaymentRepo paymentRepo) {
+        this.paymentRepo = paymentRepo;
+    }
+
+    String fetchJson(String userName) {
```

Comment:
Can this be private?

Reply:
dont think so since we need it for tests

Reply:
@pYTR0N agreed. We can change privacy settings at a future point. At the moment i would keep them like they are.